addons/udes_stock_refactoring/models/stock_picking.py

```diff
@@ -43,6 +41,43 @@ def _remove_misleading_values(self, **kwargs):
                 if current_value and current_value != value:
                     self[field] = False
 
+    @api.model
+    def _setup_refactored_picking_package_levels(self, picking, move_lines):
```

Comment:
From what i see you are not using self, and you are on stock picking class. I would suggest to define method:
`def _setup_refactored_picking_package_levels(self, move_lines):`  -- where self is the picking, maybe can add ensure one on self and below when you call to call like: `picking._setup_refactored_picking_package_levels(move_lines)`

Reply:
Good spot, problem with being lazy and copying `_new_picking_for_group`.  Will amend.

Reply:
Fixed.